src/cuda/cuda_sampling.cu

```diff
@@ -528,13 +547,20 @@ void LaunchSampleKernel(SamplingData* data, cudaStream_t stream, float* scores,
   SampleKernel<256><<<grid, block, 0, stream>>>(data->prefix_sums_adjusted.get(), indices, index_out, sample_range, indices_stride, data->thresholds.get());
 }
 
-void LaunchSort(SamplingData* data, cudaStream_t stream, float* scores_in, float* scores_out, int* indices_out, int vocab_size, int batch_size) {
+void LaunchSort(SamplingData* data, cudaStream_t stream, float* scores_in, float* scores_out, int* indices_out, int vocab_size, int batch_size, int k, float temperature) {
   // Sort indices and scores
   LaunchPopulateOffsets(data->offsets.get(), vocab_size, batch_size, stream);
   LaunchPopulateIndices(data->indices_in.get(), vocab_size, batch_size, stream);
   LaunchSortPairs<float>(data->temp_buffer.get(), data->temp_storage_bytes, scores_in, scores_out,
                          data->indices_in.get(), indices_out, vocab_size * batch_size, batch_size, data->offsets.get(),
                          stream, /*is_descending*/ true);
+
+  if (temperature != 1.f) {
+    // TODO: Fuse the temperature scaling with subsequent Softmax kernel - currently it isn't a very costly
+    // kernel, so keep it unfused
+    // We only need to scale the "top_k" elements from the sorted array (which is the output of the above kernel)
+    LaunchTemperatureScaling(scores_out, temperature, k, stream);
```

Comment:
Can we have same param order to match with `LaunchSort`? like `score_out, stream, k, temperature`?

Reply:
Sure, fixed the ordering